drivers/android/allocation.rs

```diff
@@ -37,9 +41,16 @@ impl<'a> Allocation<'a> {
             pages,
             allocation_info: None,
             free_on_drop: true,
+            file_list: List::new(),
         }
     }
 
+    pub(crate) fn take_file_list(&mut self) -> List<Box<FileInfo>> {
+        let mut list = List::new();
+        swap(&mut self.file_list, &mut list);
+        list
```

Comment:
You can use `mem::replace(&mut self.file_list, List::new())` here.

Reply:
Oh, that's nicer indeed, thanks!